lib/Menu.rb

```diff
@@ -0,0 +1,45 @@
+class Menu
+  attr_reader :meal_options, :dishes, :customer_total, :store_order
+
+  MEAL_OPTIONS = {:Hamburger=>5, :Pizza=>6, :Pasta=>4,:Fish_and_chips=>5,:Salad=>3,:Lamb_Shank=>7}
+
+    def initialize(printer = Print.new, bill = BillCalculator.new(MEAL_OPTIONS))
+      @dishes = dishes
+      @printer = printer
+      @customer_total
+      @bill = bill
+    end
+
+    def select(dishes)
+      raise 'Selection must be entered in the format ":dish=>quantity"' if dishes.class != Hash
+      raise 'Quantity must be an integer e.g. 4' if !dishes.values.any?(Integer)
+      raise 'Dish selected must be a symbol e.g. :Hamburger' if !dishes.keys.any?(Symbol)
+      @dishes = dishes
+      @bill.calculate_total(@dishes)
+
+    end
+
+    def print_all
+      @printer.print_dishes(MEAL_OPTIONS)
+    end
+
+    def my_total(total)
+      @customer_total = total
```

Comment:
Does the `@customer_total` need to be an instance variable in this class or can it be a local variable that is passed to the `total_error` method as an input parameter?

Generally variables should have the smallest scope possible (local scope < instance variable scope < global variable scope). The reason for this is because it's easier to reason about smaller scope and therefore eliminates potential bugs. Local scope is small, and by looking at a single method you can see the entire scope of the variable. An instance variable could be accessed by all the methods of the class and potentially by another class, so there is more to reason about for instance variables.

Reply:
Are there other variables that don't need to be instance variables in your code?